enhancements/baremetal/baremetal-provisioning-config.md

```diff
@@ -0,0 +1,169 @@
+created within the operator scope.
+
+This new CR would consist of the following:
+
+1. ProvisioningInterface : This is the interface name on the underlying baremetal
+server which would be physically connected to the provisioning network. This
+configuration is needed only for the underlying provisioning service (Ironic)
+and could have values like "eth1" or "ens3".
+
+2. ProvisioningIP : This is the IP address used to to bring up a NIC on the
+baremetal server for provisioning. This is also a value that is useful just to the
+provisioning service. This value should not be in the DHCP range and should not
+be used in the provisioning network for any other purpose (should be a free IP
+address.) It is expected to be provided in the format : 10.1.0.3.
+
+3. ProvisioningNetworkCIDR : This is the provisioning network with its CIDR. The
+ProvisioningIP and the ProvisioningDHCPRange are expected to be within this network.
+The value for this config is expected to be provided in the format : 10.1.0.0/24.
+
+4. ProvisioningDHCPRange : This is a range of IP addresses from which baremetal
+hosts can be assigned their IP address. This range should be provided as two
+comma seperated IP addresses where the addresses represent a contiguous range of
+IPs that are available to be allocated. It is expected to be provided in the format :
+10.1.0.10, 10.1.0.100.
+
+### User Stories [optional]
+
+This enhancement will allow users to have their Barametal IPI deployments
+customizable to their specific hardware and network requirements.
+
+This also allows IPI baremetal deployments to take place without manual
+workarounds like creating a ConfigMap for the config (which is the
+current approach we are using to test baremetal deployments.) 
+
+The new config items would be set by the installer and will be used by
+the machine-api-operator (MAO) to generate more config items that are
+derivable from these basic parameters. Put together, these config
+parameters are passed in as environment variables to the various containers
+that make up a metal3 baremetal IPI deployment. 
+
+## Design Details
+
+### Test Plan
+
+The test plan should involve making sure the openshift/installer generates
+all configuration items within the BaremetalPlatformStatus when the platform
+type is Baremetal.
+
+MAO reads this configuration and uses these to derive additional configuration
+required to bring up a metal3 cluster. E2e testing should make sure that MAO
+is able to bring up a metal3 cluster using config from this new Metal3Controller
+CR which has operator scope.
+
+### Upgrade / Downgrade Strategy
+
+Baremetal Platform type will be available for customers to use for the first
+time in Openshift 4.3. And, when it is installed, it will always start as a
+fresh baremetal installation at least in 4.3. There is no use case where a 4.2
+installation would be upgraded to a 4.3 installation with Baremetal Platform
+support enabled.
```

Comment:
As mentioned before since this slipped into 4.4 I'm not sure if we need reword some of this, perhaps we can get guidance from the review team on that.

Reply:
Yes, updated this section to make things a bit clearer based on recent discussions.